### PR TITLE
Include Meteor version details in MongoDB handshake

### PR DESCRIPTION
**Pull request by alexbevi**
The PR incorporates MongoDB's [wrapping client library](https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst#supporting-wrapping-libraries) specification for the connection handshake to allow library details to be included in the metadata written to `mongos` or `mongod` logs.

For example, this change would allow server-side logs such as the following:

> {"t":{"$date":"2024-01-27T23:10:40.108Z"},"s":"I","c":"NETWORK","id":51800,"ctx":"conn16235","msg":"client metadata","attr":{"remote":"127.0.0.1:1094","client":"conn16235","doc":{"driver":{`"name":"nodejs|Meteor"`,`"version":"4.16.0|METEOR_VERSION"`},"platform":"Node.js v18.18.2, LE","os":{"name":"linux","architecture":"x64","version":"5.15.133+","type":"Linux"}}}}

For anyone hosting clusters with connections coming from different applications this can help differentiate connections and facilitate log analysis.

TODO
- [x] Add proper version Meteor version tracking (I'm not sure where to pull this from)

### CODE REVIEW

**Comment by alexbevi** [comment]
@leonardoventurini FYI

**Comment by alexbevi** [comment]
For reference, [mongoose does the same thing](https://github.com/Automattic/mongoose/blob/7bc5395b1ade2e68c2fce4b92fa33009be0e210b/lib/drivers/node-mongodb-native/connection.js#L294-L299) so just sort of copied it over from there.

File: packages/mongo/mongo_driver.js [hunk 1]
```diff
@@ -191,6 +191,11 @@ MongoConnection = function (url, options) {
   self._oplogHandle = null;
   self._docFetcher = null;
 
+  mongoOptions.driverInfo = {
+    name: 'Meteor',
+    version: "METEOR_VERION" // TODO: however the package version needs to be included
```
[the remark this answers is not in the document]
Comment (alexbevi) [reply packages/mongo/mongo_driver.js]: Thanks @JorgenVatle, that works perfectly!

**Comment by StorytellerCZ** [comment]
Personally I love little additions like these.

**Review by StorytellerCZ — APPROVED** [review]

**Comment by alexbevi** [comment]
@StorytellerCZ I'm not sure of the review process here so just wanted to double check if I had provided everything that was needed. If there's anything else to do to get this merged let me know :)

**Comment by alexbevi** [comment]
@nachocodoner or @StorytellerCZ is there anything we can do to get this merged?

**Comment by leonardoventurini** [comment]
Hey @alexbevi we will validate and get it merged ASAP. Thanks! Sorry for taking so long to check it

**Review by leonardoventurini — APPROVED** [review]

**Pull request merged**